src/main/java/me/andrz/jackson/JsonReferenceProcessor.java

```diff
@@ -242,12 +259,26 @@ public JsonNode get(JsonReference ref) throws IOException {
         return referencedNode;
     }
 
+    private ConcurrentHashMap<Object, JsonNode> cache = new ConcurrentHashMap<>(1);
+
     public JsonNode read(URL url) throws IOException {
-        return mapper.readTree(url);
+        putIntoCache(url);
```

Comment:
@slavaschmidt Let's make the caching configurable via a boolean field `cache` on `JsonReferenceProcessor` with a default of `true`.